packages/permissions/src/utils/has-permissions.ts

```diff
@@ -237,19 +240,6 @@ const getDataFenceByPermissionGroup = (
 };
 
 export const hasAppliedDataFence = (options: TOptionsForAppliedDataFence) => {
-  // If user has `*Manage` as their `appliedPermission`, we bypass `dataFence`.
-  const hasAppliedManagePermission = options.demandedPermissions.some(
```

Comment:
because we fall back now to general permissions if dataFence permissions are not met, we do not need to check if the use has `manage` general permission first INSIDE `hasAppliedDataFence`, this simplifies the logic further and makes the flow for permission check straight forward: 

1- check if the user meets the dataFence permissions
2- if not, check if he meets the general permissions + action rights